web/src/emoji_picker.ts

```diff
@@ -472,36 +502,42 @@ function change_focus_to_filter() {
     reset_emoji_showcase();
 }
 
-export function navigate(event_name, e) {
+export function navigate(event_name: string, e?: JQuery.KeyDownEvent): boolean {
     if (
         event_name === "toggle_reactions_popover" &&
         is_open() &&
-        (search_is_active === false || search_results.length === 0)
+        (!search_is_active || search_results.length === 0)
     ) {
         hide_emoji_popover();
         return true;
     }
 
     // If search is active and results are empty then return immediately.
-    if (search_is_active === true && search_results.length === 0) {
+    if (search_is_active && search_results.length === 0) {
         // We don't want to prevent default for keys like Backspace and space.
         return false;
     }
 
     if (event_name === "enter") {
+        assert(e !== undefined);
+        assert(e.target instanceof HTMLElement);
```

Comment:
Do we have another way to deal with this. Using `assert` made sense to me. Added comment explaining the reason though.

Reply:
This is probably OK but there's probably a refactor of the caller(s) that would make this unnecessary.